app/controllers/favorite_items_controller.rb

```diff
@@ -0,0 +1,29 @@
+class FavoriteItemsController < ApplicationController
+  layout 'user'
+
```

Comment:
require the current_user before

def require_user
    unless current_user
      flash[:info] = "This is reserved for members, please login or register fisrt!"
      redirect_to root_path
    end
  end